components/dropdown/basic/src/index.js

```diff
@@ -102,68 +82,53 @@ class DropdownBasic extends Component {
   /**
    * Document click event handler.
    */
-  _onDocumentClick = event => {
+  const onDocumentClick = event => {
     const {target} = event
-    const isClickOutsideDropdown = !this._wrapper.current.contains(target)
+    const isClickOutsideDropdown = !wrapper.current.contains(target)
 
-    isClickOutsideDropdown &&
-      this.props.closeOnDocumentClick &&
-      this._toggleMenu(false)
+    isClickOutsideDropdown && props.closeOnDocumentClick && toggleMenu(false)
   }
 
   /**
    * Document listener handler.
    */
-  _handleListener(add) {
+  const handleListener = add => {
     const listenerMethod = add ? 'addEventListener' : 'removeEventListener'
-    document[listenerMethod]('click', this._onDocumentClick, false)
-  }
-
-  componentDidMount() {
-    this.state.expanded && this._handleListener(true)
-  }
-
-  componentWillUnmount() {
-    this._handleListener(false)
+    document[listenerMethod]('click', onDocumentClick, false)
   }
-
-  render() {
-    const {expanded, collapseByTouch} = this.state
-    const {button, menu, expandOnMouseOver} = this.props
-    const {text, icon: Icon} = button
-    const ArrowButtonIcon = button.arrowButtonIcon || Chevronbottom
-    const wrapperClassName = cx('sui-DropdownBasic', {
-      'is-expanded': expanded
-    })
-    return (
-      <div
-        className={wrapperClassName}
-        onMouseOver={expandOnMouseOver ? this._onMouseOver : this._doNothing}
-        onMouseOut={expandOnMouseOver ? this._onMouseOut : this._doNothing}
-      >
-        <div className="sui-DropdownBasic-buttonWrap">
-          <button
-            className="sui-DropdownBasic-button"
-            onClick={expandOnMouseOver ? this._doNothing : this._toggleMenu}
-            onTouchStart={
-              expandOnMouseOver && collapseByTouch
-                ? this._toggleMenu
-                : this._doNothing
-            }
-          >
-            <div className="sui-DropdownBasic-buttonContent">
-              {Icon && <Icon svgClass="sui-DropdownBasic-buttonIcon" />}
-              <span>{text}</span>
-            </div>
-            <ArrowButtonIcon svgClass="sui-DropdownBasic-buttonIcon" />
-          </button>
-        </div>
-        <div className="sui-DropdownBasicMenu" ref={this._wrapper}>
-          {menu.map(this._renderMenuItem)}
-        </div>
+  const {button, menu, expandOnMouseOver} = props
+  const {text, icon: Icon} = button
+  const ArrowButtonIcon = button.arrowButtonIcon || Chevronbottom
+  const wrapperClassName = cx('sui-DropdownBasic', {
+    'is-expanded': expanded
+  })
+
+  return (
+    <div
+      className={wrapperClassName}
+      onMouseOver={expandOnMouseOver ? onMouseOver : NO_OP}
+      onMouseOut={expandOnMouseOver ? onMouseOut : NO_OP}
+    >
+      <div className="sui-DropdownBasic-buttonWrap">
```

Comment:
Can you create a `base-class` for "sui-DropdownBasic"?

Reply:
I didn't want to touch internals but I will recheck. ;) Thanks!